nym-vpn-app/src-tauri/src/grpc/client.rs

```diff
@@ -59,13 +60,31 @@ pub enum VpndError {
 }
 
 #[derive(Debug, Default, Clone)]
-pub struct GrpcClient(Transport);
+pub struct GrpcClient {
+    transport: Transport,
+    user_agent: UserAgent,
+}
 
 impl GrpcClient {
     #[instrument(skip_all)]
-    pub fn new(config: &AppConfig, cli: &Cli) -> Self {
-        let client = GrpcClient(Transport::from((config, cli)));
-        match &client.0 {
+    pub fn new(config: &AppConfig, cli: &Cli, pkg: &PackageInfo) -> Self {
+        let git_commit = crate::build_info()
+            .version_control
+            .as_ref()
+            .and_then(|vc| vc.git())
+            .map(|g| g.commit_short_id.clone())
+            .unwrap_or_default();
+
+        let client = GrpcClient {
+            transport: Transport::from((config, cli)),
+            user_agent: UserAgent {
+                application: pkg.name.clone(),
+                version: pkg.version.to_string(),
```

Comment:
It's pretty useful to have the daemon version here as well in addition to the app version. See e.g how it's done for nym-vpnc. Same for the git commit, although it's probably less important in comparison

Reply:
Checked the code of vpnc, at first glance it looks a bit dirty to have to call grpc `Info` method before any other method needing the user_agent, just in the purpose to assemble the version value.

Imo this logic should be handled in daemon side only, e.g the daemon mutates `user_agent.version` by appending its version to the value or we add a new field for it like `user_agent.daemon_version`.
As well this would remove the possibility for grpc clients to set 'fake' daemon version

Reply:
The proper solution imho would be to introduce a new type on the lib side with additional fields. But then this needs to be compatible with the existing UserAgent type since this is used in crates in the monorepo so probably need to create some shared trait in `nym-http-api-client`

But if we're not doing that, then the simpler and less error prone way I think is to set it all up front in once place rather than some field getting rewritten half way down the call stack

At the end of the day I think the first priority here is to just set them consistently according to the format in the ticket. Feel free to try to refactor things if you think you have the extra time

Reply:
I think relying on clients to set user_agent daemon info is wrong.
Also with the current solution each client would have to carry on this extra logic (calling rpc `Info` to fill user agent with daemon info before calling any other rpc methods requiring that user agent).

>  I think is to set it all up front in once place rather than some field getting rewritten half way down the call stack

So yea this "once place" would be daemon instead of a piece of logic repeated for each client implementation (vpnc, vpn-cli, vpn-app)

Anyway, for the sake of 'moving on', and not falling into a consequent refactor, I implemented the logic in the tauri app as well https://github.com/nymtech/nym-vpn-client/pull/1236/commits/62d45f6bca6e328f5c815ee972fdbc57415cb713